addons/mail/models/discuss/discuss_channel.py

```diff
@@ -633,6 +633,21 @@ def _message_post_after_hook(self, message, msg_vals):
         self._set_last_seen_message(message)
         return super()._message_post_after_hook(message, msg_vals)
 
+    def _process_attachments_for_post(self, attachments, attachment_ids, message_values):
+        return_values = super()._process_attachments_for_post(attachments, attachment_ids, message_values)
+
+        if attachments:
+            for attachment, m2m_command in zip(attachments, return_values.get('attachment_ids', [])):
+                if len(attachment) == 3:
+                    _, _, info = attachment
+                    voice = info and info.get('voice')
+                    if voice:
+                        # Extract attachment_id from the (4, attachment_id) M2M command
+                        attachment_id = m2m_command[1] if m2m_command[0] == 4 else None
+                        if attachment_id:
+                            self.env['ir.attachment'].browse(attachment_id).exists()._post_add_create(voice=True)
```

Comment:
Looking for closely, I don't think it is the right place to put the `_post_add_create`.

If the attachments are already created, the "voice" (through `_post_add_create`) should be set at their creation, not when they are linked to a message.

Reply:
The attachments are created in `_process_attachments_for_post` in the `mail.thread` but as the voice is coming from the discuss channel we have created a voice here.

Reply:
> Where is the info voice key coming from originally?

The voice is coming form the whatsapp side with in he webhook. See https://github.com/odoo/enterprise/pull/63075
Whenever user is sending audio message from Whatsapp, we will receive voice in webhook.
In the flow, we will extract information from the webhook and will post it in active channel using `message_post`.
![image](https://github.com/user-attachments/assets/456b1cb0-0fb5-4ce3-a9a2-0319e2ee479c)